pkg/registry/client.go

```diff
@@ -14,30 +14,86 @@
 package registry
 
 import (
-	"net/http"
-	"strings"
-
-	"github.com/heroku/docker-registry-client/registry"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
 	"github.com/okteto/okteto/pkg/log"
+	"github.com/okteto/okteto/pkg/okteto"
 )
 
-// NewRegistryClient creates a new Registry with the given URL and credentials, then Ping()s it
-// before returning it to verify that the registry is available.
-func NewRegistryClient(registryURL, username, password string) (*registry.Registry, error) {
-	transport := http.DefaultTransport
-	return newFromTransport(registryURL, username, password, transport)
+type OktetoRegistryAuthenticator struct {
+	username string
+	password string
+}
+
+func (a *OktetoRegistryAuthenticator) Authorization() (*authn.AuthConfig, error) {
+	return &authn.AuthConfig{
+		Username: a.username,
+		Password: a.password,
+	}, nil
+}
+
+func newBasicAuthRegistry(username, password string) *OktetoRegistryAuthenticator {
```

Comment:
Don't you need to export `newBasicAuthRegistry`? If not, maybe `OktetoRegistryAuthenticator` can be local to the package